gen/field.go

```diff
@@ -30,6 +30,7 @@ var reservedIdentifiers = map[string]struct{}{
 	"ToWire":   {},
 	"FromWire": {},
 	"String":   {},
+	"Equals":   {},
```

Comment:
Discussed offline with @abhinav. This would get rejected at the parser level itself.